Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -19,121 +19,160 @@ class VirtualMachineSlot: NSObject, ObservableObject {
-                self.status = .running(launchConfiguration, .any)
+            let startTask = Task { try await createManagedVm(launchConfiguration: launchConfiguration) }
+            self.status = .starting(launchConfiguration, startTask)
+            let newVM = try await startTask.value
+            if startTask.isCancelled {
+                Logger.helper.debug("Start task was cancelled for \(newVM.handle).")
+                throw Errors.vmStartCancelled
             }
+
+            Logger.helper.log("Setting \(role.displayName) slot to running \(newVM.handle).")
+            newVM.machine.delegate = self
+            self.status = .running(newVM)
         } catch {
             Logger.helper.error("Error launching VM: \(error.localizedDescription)")
-            Logger.helper.error("Attempting Cleanup of \(launchConfiguration.handle)")
-
-            do {
-                try await vmManager.removeVM(name: launchConfiguration.handle)
-            } catch {
-                Logger.helper.error("Failed to remove VM file: \(error)")
-            }
-            self.managedVirtualMachine = nil
-            self.status = .crashed(error)
+            try? await stop(withError: error)
             throw error
         }
     }
 
-    @MainActor
-    func stopVirtualMachine() async throws {
-        switch status {
-        case .starting, .running:
-            self.status = .stopping
-        default:
-            break
+    private func createManagedVm(launchConfiguration: LaunchConfiguration) async throws -> ManagedVirtualMachine {
+        Logger.helper.log("Creating VM \(launchConfiguration.handle).")
+        let virtualMachine = try await launchConfiguration.setupVirtualMachine()
+        try await virtualMachine.start()
+
+        let ipAddress: IPv4Address
+        if launchConfiguration.waitForNetworking {
+            ipAddress = try await vmManager.ipAddress(forVmWithName: launchConfiguration.handle)
+            Logger.helper.log("Startup complete – IP Address: \(ipAddress.debugDescription).")
+        } else {
+            Logger.helper.log("Startup in progress – skipped waiting for IP address per launch configuration.")
+            ipAddress = .any
         }
-        if let machine = managedVirtualMachine?.machine, machine.canStop {
-            do {
-                try await machine.stop()
-            } catch {
-                Logger.helper.error("Failed to stop VM: \(error)")
+        return ManagedVirtualMachine(machine: virtualMachine, config: launchConfiguration, ip: ipAddress)
+    }
+
+    private func stopManagedVm(_ mvm: ManagedVirtualMachine) async {
+        Logger.helper.log("Stopping VM \(mvm.handle).")
+        /// Quit responding to delegate methods
+        mvm.machine.delegate = nil
+        do {
+            if mvm.machine.canStop {
+                try await mvm.machine.stop()
+                Logger.helper.log("Stopped VM \(mvm.handle).")
             }
```

Comment:
Would it be useful to log in case we try to stop a VM that can't be stopped?

```suggestion
            guard mvm.machine.canStop else {
                Logger.helper.log("Attempted to stop VM \(mvm.handle), but it cannot be stopped.")
                return 
            }
            
            try await mvm.machine.stop()
            Logger.helper.log("Stopped VM \(mvm.handle).")
```

Reply:
We can probably deduce it, but given that #128 isn't resolved and we're on a logging-spree to increase stability, I'll add one. 👍

Reply:
Handled in ffc70d0f848d3d525007bd7721c92935dfc09586.